### ISSUE TITLE
duplicate_enum_cases and conditional compilation

### ISSUE THREAD

**Issue by nevil** [realm/SwiftLint#2782]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The newly added `duplicate_enum_cases` does not consider conditional compilation flags.
It reports duplicates even though there are none in the configuration actually compiled.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint duplicate_enum_cases_if_else.swift
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths duplicate_enum_cases_if_else.swift
Linting 'duplicate_enum_cases_if_else.swift' (1/1)
duplicate_enum_cases_if_else.swift:3:6: error: Duplicate Enum Cases Violation: Enum can't contain multiple cases with the same name. (duplicate_enum_cases)
duplicate_enum_cases_if_else.swift:5:6: error: Duplicate Enum Cases Violation: Enum can't contain multiple cases with the same name. (duplicate_enum_cases)
duplicate_enum_cases_if_else.swift:7:6: error: Duplicate Enum Cases Violation: Enum can't contain multiple cases with the same name. (duplicate_enum_cases)
Done linting! Found 3 violations, 3 serious in 1 file.
```

### Environment

0.33.0
Cocoapods

```swift
enum Tag: String {
#if CONFIG_A
case value = "CONFIG_A"
#elseif CONFIG_B
case value = "CONFIG_B"
#else
case value = "CONFIG_DEFAULT"
#endif
}
```

**Comment by a-25** [comment]
@marcelofabri: I debugged this and found out that SourceKit gives all enum elements regardless of conditional compilation. Seems that your fix https://bugs.swift.org/browse/SR-9630 is going to fix this bug also.
So I suggest to close this issue until Swift 5.1 (like this one https://github.com/realm/SwiftLint/issues/2542).

**Comment by marcelofabri** [comment]
@a-25 I don't think that SR-9630 is related to this: that ticket is about having a better way to determine which is the current Swift version.

**Comment by kylebrowning** [comment]
This is still happening just FYI.

**Comment by mwilson-within** [comment]
This issue still exists in 0.41.0.

**Comment by ppamorim** [comment]
Still happening on 0.46.1

**Comment by jonnyijapan** [comment]
This is still a bug.

**Comment by kylebrowning** [comment]
Yeah we moved on to another linter.

**Comment by jonnyijapan** [comment]
I'll double check with the latest version though. Cause seems there is one. 😀

**Comment by jonnyijapan** [comment]
OK, this issue is fixed at least in 0.52.2 and probably since a few versions earlier
